mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -1118,8 +1118,14 @@ static SafeSocketHandle Accept_internal (SafeSocketHandle safeHandle, out int er
 
 #region Bind
 
+#if FEATURE_NO_BSD_SOCKETS
+		[Obsolete  ("System.Net.Sockets.Socket:Bind is not supported on this platform.")]
+#endif
```

Comment:
As Sebastien mentioned on Slack, the Obsolete attribute is the only way for developers to be notified of API that doesn't work at build time (instead of runtime).